app/tests/spec/lib/validate.js

```diff
@@ -134,5 +134,112 @@ define(function (require, exports, module) {
         assert.isTrue(Validate.isPasswordValid(createRandomHexString(Constants.PASSWORD_MIN_LENGTH)));
       });
     });
+
+    describe('isDataValid', function () {
+      /*eslint-disable sorting/sort-object-props */
+
+      it('is defined', function () {
+        assert.isFunction(Validate.isDataValid);
+      });
+
+      it('expects two arguments', function () {
+        assert.lengthOf(Validate.isDataValid, 2);
+      });
+
+      it('returns true when simple type matches the schema', function () {
+        assert.isTrue(Validate.isDataValid('foo', 'String'));
+      });
+
+      it('returns false when simple type does not match the schema', function () {
+        assert.isFalse(Validate.isDataValid('foo', 'Number'));
+      });
+
+      it('handles null', function () {
+        assert.isTrue(Validate.isDataValid(null, 'Null'));
+      });
+
+      it('handles undefined', function () {
+        assert.isTrue(Validate.isDataValid(undefined, 'Undefined'));
+      });
+
+      it('handles false', function () {
+        assert.isTrue(Validate.isDataValid(false, 'Boolean'));
+      });
+
+      it('handles zero', function () {
+        assert.isTrue(Validate.isDataValid(0, 'Number'));
+      });
+
+      it('handles empty string', function () {
+        assert.isTrue(Validate.isDataValid('', 'String'));
+      });
+
+      it('returns true when complex type matches the schema', function () {
+        assert.isTrue(Validate.isDataValid(
+          [ 'foo', 3.14159265359, {} ],
+          [ 'String', 'Number', 'Object' ]
+        ));
+      });
+
+      it('returns false when complex type does not match the schema', function () {
+        assert.isFalse(Validate.isDataValid(
+          [ 'foo', 3.14159265359, [] ],
+          [ 'String', 'Number', 'Object' ]
```

Comment:
> In IRC you mentioned that perhaps we'd choose to handle that by padding the schema array instead. I figured it made sense to make wait until we are validating that property before making the decision.

Without a wrapper, padding won't work very well in practice, because we'd have to analyze the possible input, then create a schema at runtime based on the input, which is backwards to how it's supposed to work. Which makes the second comment (below) seem like a winner.

> Just remembered, the other approach I had in mind for this was a separate function, isArrayValid, which does the padding and then calls isDataValid to perform the actual validation. Mentioning it here in case it is of interest later.

 Can you make a proposal how we'd handle [sending sync data](https://github.com/mozilla/fxa-content-server/blob/2d7c9e2afe7ad9df0f721ba527615f5b420521ea/app/scripts/models/auth_brokers/fx-sync.js#L207-L221) because that's going to be the next message that's validated.